pkg/cmd/util/clientcmd/factory.go

```diff
@@ -1,715 +1,117 @@
-			return ns, true, nil
-		}
-	}
-
-	return api.NamespaceDefault, false, nil
-}
-
-// ConfigAccess implements ClientConfig
-func (c defaultingClientConfig) ConfigAccess() kclientcmd.ConfigAccess {
-	return c.nested.ConfigAccess()
-}
-
-type errConfigurationMissing struct {
-	err error
-}
-
-func (e errConfigurationMissing) Error() string {
-	return fmt.Sprintf("%v", e.err)
-}
-
-func IsConfigurationMissing(err error) bool {
-	switch err.(type) {
-	case errConfigurationMissing:
-		return true
-	}
-	return kclientcmd.IsContextNotFound(err)
-}
-
-// ClientConfig returns a complete client config
-func (c defaultingClientConfig) ClientConfig() (*restclient.Config, error) {
-	cfg, err := c.nested.ClientConfig()
-	if err == nil {
-		return cfg, nil
-	}
-
-	if !kclientcmd.IsEmptyConfig(err) {
-		return nil, err
-	}
-
-	// TODO: need to expose inClusterConfig upstream and use that
-	if icc, err := restclient.InClusterConfig(); err == nil {
-		glog.V(4).Infof("Using in-cluster configuration")
-		return icc, nil
-	}
-
-	return nil, errConfigurationMissing{fmt.Errorf(`Missing or incomplete configuration info.  Please login or point to an existing, complete config file:
-
-  1. Via the command-line flag --config
-  2. Via the KUBECONFIG environment variable
-  3. In your home directory as ~/.kube/config
-
-To view or setup config directly use the 'config' command.`)}
-}
-
 // Factory provides common options for OpenShift commands
 type Factory struct {
-	*cmdutil.Factory
-	OpenShiftClientConfig kclientcmd.ClientConfig
-	clients               *clientCache
-
-	ImageResolutionOptions FlagBinder
-
-	PrintResourceInfos func(*cobra.Command, []*resource.Info, io.Writer) error
+	ClientAccessFactory
+	kcmdutil.ObjectMappingFactory
+	kcmdutil.BuilderFactory
 }
 
-func DefaultGenerators(cmdName string) map[string]kubectl.Generator {
-	generators := map[string]map[string]kubectl.Generator{}
-	generators["run"] = map[string]kubectl.Generator{
-		"deploymentconfig/v1": deploycmd.BasicDeploymentConfigController{},
-		"run-controller/v1":   kubectl.BasicReplicationController{}, // legacy alias for run/v1
-	}
-	generators["expose"] = map[string]kubectl.Generator{
-		"route/v1": routegen.RouteGenerator{},
-	}
-
-	return generators[cmdName]
-}
+var _ kcmdutil.Factory = &Factory{}
 
 // NewFactory creates an object that holds common methods across all OpenShift commands
-func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
-	restMapper := registered.RESTMapper()
-
-	clients := &clientCache{
-		clients: make(map[string]*client.Client),
-		configs: make(map[string]*restclient.Config),
-		loader:  clientConfig,
-	}
-
-	w := &Factory{
-		Factory:                cmdutil.NewFactory(clientConfig),
-		OpenShiftClientConfig:  clientConfig,
-		clients:                clients,
-		ImageResolutionOptions: &imageResolutionOptions{},
-	}
+func NewFactory(optionalClientConfig kclientcmd.ClientConfig) *Factory {
+	// clients := &clientCache{
```

Comment:
@deads2k @fabianofranz please review this commit (I'll remove the commented out lines below later)